### ISSUE TITLE
What does @layer_register(log_shape=True) do before a layer?

### ISSUE THREAD

**Issue by john81923** [tensorpack/tensorpack#1032]
As the title mentioned, I'm curious about what @layer_register(log_shape=True) do before a layer?

such as 
`@layer_register(log_shape=True)`
`def Conv2D(x, out_channel, kernel_shape,
           padding='SAME', stride=1,
           W_init=None, b_init=None,
           nl=tf.nn.relu, split=1, use_bias=True):`
